src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -726,84 +726,29 @@ private void requestMultipleUpload(Intent data, int resultCode) {
     }
 
 
-    private void requestSimpleUpload(Intent data, int resultCode) {
-        String filePath = null;
-        String mimeType = null;
+    private void requestSimpleUpload(final Intent data, int resultCode) {
 
-        Uri selectedImageUri = data.getData();
-
-        try {
-            mimeType = getContentResolver().getType(selectedImageUri);
-
-            String fileManagerString = selectedImageUri.getPath();
-            String selectedImagePath = UriUtils.getLocalPath(selectedImageUri, this);
-
-            if (selectedImagePath != null)
-                filePath = selectedImagePath;
-            else
-                filePath = fileManagerString;
-
-        } catch (Exception e) {
-            Log_OC.e(TAG, "Unexpected exception when trying to read the result of " +
-                    "Intent.ACTION_GET_CONTENT", e);
+        int behaviour = (resultCode == UploadFilesActivity.RESULT_OK_AND_MOVE) ? FileUploader.LOCAL_BEHAVIOUR_MOVE :
+                FileUploader.LOCAL_BEHAVIOUR_COPY;
 
-        } finally {
-            if (filePath == null) {
-                Log_OC.e(TAG, "Couldn't resolve path to file");
-                Toast t = Toast.makeText(
-                        this, getString(R.string.filedisplay_unexpected_bad_get_content),
-                        Toast.LENGTH_LONG
-                );
-                t.show();
-                return;
-            }
-        }
+        ArrayList<Parcelable> mStreamsToUpload = new ArrayList<Parcelable>() {{
```

Comment:
Besides, we should move this code out of the method, or change it to receive directly an ``ArrayList<Parcelable>``` object instead of an ```Intent data```. See the caller code; in some situations ```requestSimpleUpload``` is called in a loop, what will result in a several ```UriUploader``` objects created and, more problematic, in several ```CopyAndUploadContentUrisTask```